### PR TITLE
UI touchups for the route drag/drop

### PR DESCRIPTION
**Pull request by michaelkirk**
Several small changes - probably easiest to understand in the individual commits.

**before**
Screen Shot 2021-09-09 at 3 43 43 PM

**after**
Screen Shot 2021-09-09 at 3 41 20 PM

https://user-images.githubusercontent.com/217057/132773537-d0fddc9a-76d5-410f-a34b-37093ed88284.mp4

### CODE REVIEW

File: widgetry/src/widgets/drag_drop.rs [hunk 1]
```diff
@@ -329,6 +337,17 @@ impl<T: 'static + Copy + PartialEq> WidgetImpl for DragDrop<T> {
             self.state = new_state;
             self.recalc_draw(ctx);
         }
+
+        match self.state {
```
Comment (michaelkirk) [inline widgetry/src/widgets/drag_drop.rs:341]: The 3 minutes I spent doing this was definitely the biggest bang/buck.

File: game/src/common/waypoints.rs [hunk 2]
```diff
@@ -96,18 +110,19 @@ impl InputWaypoints {
 
             delete_buttons.push(
                 ctx.style()
-                    .btn_plain_destructive
-                    .text("X")
+                    .btn_close()
+                    .override_style(&ctx.style().btn_plain_destructive)
```
Comment (michaelkirk) [inline game/src/common/waypoints.rs:114]: usually we build the button *from* the style... but some of these "special buttons" like the close button make that a bit awkward, so I added this override style method.

The alternative would be to do something like:

ctx.style().btn_plain_destructive.btn_image(widgetry::include_image_bytes("../../some/hardcoded/path/into/an/external/crate"))

File: game/src/common/waypoints.rs [hunk 3]
```diff
@@ -69,22 +69,22 @@ impl InputWaypoints {
         let mut delete_buttons = Vec::new();
 
         for (idx, waypt) in self.waypoints.iter().enumerate() {
-            let batch = Text::from(Line(format!("{}) {}", waypt.order, waypt.label))).render(ctx);
-            let bounds = batch.get_bounds();
-            let image = Image::from_batch(batch, bounds)
-                .color(RewriteColor::NoOp)
+            let waypoint = ctx
```
Comment (dabreegster) [inline game/src/common/waypoints.rs:85]: This is _much_ cleaner, thank you for fixing. Using button builders just to grab the batch makes way more sense than trying to replicate that style with lower-level APIs like Text.

File: game/src/common/waypoints.rs [hunk 4]
```diff
@@ -69,11 +69,25 @@ impl InputWaypoints {
         let mut delete_buttons = Vec::new();
 
         for (idx, waypt) in self.waypoints.iter().enumerate() {
+            let icon = {
+                let text = Text::from(Line(waypt.order.to_string()).fg(Color::WHITE).bold_body());
+                let batch = text.render(ctx);
+                let bounds = batch.get_bounds();
+                let image = Image::from_batch(batch, bounds)
```
Comment (dabreegster) [inline game/src/common/waypoints.rs:76]: Haha wow, this is an awesome way to make a circular background for text! That `CornerRounding` is magic.

File: widgetry/src/widgets/drag_drop.rs (hunk 1 above)
Comment (dabreegster) [reply widgetry/src/widgets/drag_drop.rs:341]: Nice! I'm noticing some stuttering -- do you see something like this?
screencast
It's likely due to https://github.com/a-b-street/abstreet/blob/50b8c533bfcb98d79bbb029ec444ce37e871ee17/widgetry/src/runner.rs#L39. IIRC, we reset the cursor every single event, and rely on caller code to keep setting other states. I'm not sure which event is causing this to flicker here; I can take a look tomorrow...

**Review by dabreegster — APPROVED** [review]
LGTM, thank you for polishing this up! I can dig into the cursor flicker later if you like, but I'd be fine merging this as-is.

**Comment by dabreegster** [comment]
I'm going to merge now to avoid accumulating conflicts in the route UI. I'll take a look at the flickery cursor. Thanks again for the PR!

**Comment by michaelkirk** [comment]
>  I'm going to merge now to avoid accumulating conflicts in the route UI. I'll take a look at the flickery cursor. Thanks again for the PR!

Sorry for not addressing this right away - I'm working on some other things for the next few days.

**Comment by michaelkirk** [comment]
> I'm going to merge now to avoid accumulating conflicts in the route UI. I'll take a look at the flickery cursor. Thanks again for the PR!

It seems like you've fixed this @dabreegster? I can no longer reproduce it.

**Comment by dabreegster** [comment]
> It seems like you've fixed this @dabreegster? I can no longer reproduce it.

Nope, it's still present on my end. If you're not seeing it when you slowly drag cards around, it's likely some kind of platform-dependent winit issue. I haven't dug into it further.

**Comment by michaelkirk** [comment]
Hm... I can't reproduce it on my mac w/ native, safari, nor chrome.

Can you reproduce it on the web, or just on native?

**Comment by dabreegster** [comment]
> Can you reproduce it on the web, or just on native?

Just on native (Xorg, not wayland). Web is fine.

I'd say this is low-priority, accordingly.

**Pull request merged**